syngenta_digital_dta/mongo/adapter.py

```diff
@@ -61,7 +71,12 @@ def find_one(self, **kwargs):
         return self.connection.find_one(kwargs['query'])
 
     def find(self, **kwargs):
-        results = self.connection.find(kwargs['query'])
+        if kwargs.get('page_size') and kwargs.get('page_number'):
+            skips = kwargs['page_size'] * (kwargs['page_number'] - 1)
+            results = self.connection.find(kwargs['query']).skip(skips).limit(kwargs['page_size'])
+        else:
+            results = self.connection.find(kwargs['query'])
+
```

Comment:
This isn't very readable and doesn't support the other optional parameters possible, so it's not very extensible:

```suggestion
        kwargs.pop('operation', None)
        query = kwargs.pop('query')
        results = self.connection.find(query, **kwargs)
```

Reply:
this is much better